extensions/sockets/src/main/java/org/mule/module/socket/api/connection/udp/UdpListenerConnection.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.socket.api.connection.udp;
+
+import static org.mule.module.socket.internal.SocketUtils.configureConnection;
+import static org.mule.module.socket.internal.SocketUtils.createPacket;
+import org.mule.module.socket.api.connection.ListenerConnection;
+import org.mule.module.socket.api.exceptions.ReadingTimeoutException;
+import org.mule.module.socket.api.socket.udp.UdpSocketProperties;
+import org.mule.module.socket.api.source.SocketAttributes;
+import org.mule.module.socket.internal.worker.SocketWorker;
+import org.mule.module.socket.internal.worker.UdpWorker;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.extension.api.runtime.MessageHandler;
+import org.mule.runtime.extension.api.runtime.source.Source;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.SocketTimeoutException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UdpListenerConnection extends AbstractUdpConnection implements ListenerConnection
+{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UdpListenerConnection.class);
+    private boolean wasDisconnected = false;
```

Comment:
this is like the 4th time I've seen this variable... Put it in some base class and also try to inherit its usages